modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/CopyFormInstanceMVCActionCommand.java

```diff
@@ -126,6 +151,25 @@ protected ResourceBundle getResourceBundle(Locale locale) {
 	@Reference
 	protected SaveFormInstanceMVCCommandHelper saveFormInstanceMVCCommandHelper;
 
+	private void _removeExpirationDateDDMFormFieldValue(
+		DDMFormValues settingsDDMFormValues) {
+
+		for (DDMFormFieldValue ddmFormFieldValue :
+				settingsDDMFormValues.getDDMFormFieldValues()) {
+
+			if (StringUtil.equals(
+					ddmFormFieldValue.getName(), "expirationDate")) {
+
+				ddmFormFieldValue.setValue(new UnlocalizedValue(""));
+			}
+			else if (StringUtil.equals(
+						ddmFormFieldValue.getName(), "neverExpire")) {
+
+				ddmFormFieldValue.setValue(new UnlocalizedValue("true"));
+			}
```

Comment:
Does it make sense to edit the `neverExpire` value to `true` if this method is called? 
Especially since the situation for this method to be called is when the `ddmFormInstanceSettings.neverExpire()` is `false`.

Reply:
Hi @G-Sa ,

yes, because the expected behavior when creating a copy of an expired form is to create it in the default way, which is `neverExpire` (without an expiration date).

Reply:
Makes sense! all good then